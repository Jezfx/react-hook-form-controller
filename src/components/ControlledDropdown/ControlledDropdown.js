import React from "react";
import PropTypes from "prop-types";
import { TextField, FormLabel, FormControl } from "@material-ui/core";
import { useFormContext, Controller } from "react-hook-form";

const ControlledDropdown = ({ label, options = [], name }) => {
  const { control, errors } = useFormContext();

  const error = errors[name];

  return (
    <FormControl>
      <FormLabel htmlFor={name}>{label}</FormLabel>
      <Controller
        id={name}
        name={name}
        defaultValue={false}
        control={control}
        render={({ onChange, value = "", name }) => (
          <TextField
            select
            onChange={onChange}
            value={value || ""}
            name={name}
            error={!!error}
            helperText={error ? error.message : null}
            SelectProps={{
              native: true,
              inputProps: {
                name: name,
              },
            }}
          >
            {options?.map(({ label, ...rest }, index) => (
              <option key={index} {...rest}>
                {label}
              </option>
            ))}
          </TextField>
        )}
      />
    </FormControl>
  );
};

ControlledDropdown.propTypes = {
  label: PropTypes.string,
  name: PropTypes.string,
  options: PropTypes.arrayOf(
    PropTypes.shape({
      value: PropTypes.string,
      label: PropTypes.string,
      name: PropTypes.string,
    })
  ),
};

ControlledDropdown.defaultProps = {
  options: [],
};

export default ControlledDropdown;
