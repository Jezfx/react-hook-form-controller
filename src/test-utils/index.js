import { render } from "@testing-library/react";
import { useForm, FormProvider } from "react-hook-form";

export const renderWithProviders = (ui, { defaultValues = {} } = {}) => {
  const Wrapper = ({ children }) => {
    const methods = useForm({});
    return <FormProvider {...methods}>{children}</FormProvider>;
  };
  return render(ui, { wrapper: Wrapper });
};
