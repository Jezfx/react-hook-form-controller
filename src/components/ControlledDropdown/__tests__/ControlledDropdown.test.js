import React from "react";
import ControlledDropdown from "../";
import { renderWithProviders } from "../../../test-utils";

describe("ControlledDropdown", () => {
  it("should render correctly", () => {
    const options = [
      { name: "bar", label: "Bar" },
      { name: "baz", label: "Baz" },
    ];
    const { asFragment } = renderWithProviders(
      <ControlledDropdown options={options} name="foo" label="Foo" />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
