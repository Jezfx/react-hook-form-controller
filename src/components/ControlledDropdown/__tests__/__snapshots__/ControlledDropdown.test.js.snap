// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ControlledDropdown should render correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root"
  >
    <label
      class="MuiFormLabel-root"
      for="foo"
    >
      Foo
    </label>
    <div
      class="MuiFormControl-root MuiTextField-root"
    >
      <div
        class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
      >
        <select
          aria-invalid="false"
          class="MuiSelect-root MuiSelect-select MuiInputBase-input MuiInput-input"
          name="foo"
        >
          <option
            name="bar"
          >
            Bar
          </option>
          <option
            name="baz"
          >
            Baz
          </option>
        </select>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSelect-icon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
